### PR TITLE
Remove old nose testing code

### PR DESCRIPTION
**Pull request by dstansby**
Lots of old code that I guess was used when we used `nose` for testing. I've just nuked it since it was in an underscored folder, happy to deprecate instead.

### CODE REVIEW

**Comment by dstansby** [comment]
Not sure why I closed this...

**Comment by dstansby** [comment]
Ah, I see it's scheduled for removal in `2.3` (`3.0`?). I'll leave this open and it can always go in when `2.2` is out.

File: lib/matplotlib/testing/decorators.py [hunk 1]
```diff
@@ -41,23 +41,13 @@ def _knownfailureif(fail_condition, msg=None, known_exception_class=None):
     if the exception is an instance of this class. (Default = None)
 
     """
-    if is_called_from_pytest():
```
Comment (QuLogic) [inline lib/matplotlib/testing/decorators.py]: This function is private, so we should probably just replace its callers with the direct pytest marker call.

Comment (anntzer) [reply lib/matplotlib/testing/decorators.py]: technically it is public but we should still do the switch and deprecate it.

Comment (QuLogic) [reply lib/matplotlib/testing/decorators.py]: No it isn't. The one being removed below is public, but it's already deprecated. This one is private.

Comment (QuLogic) [reply lib/matplotlib/testing/decorators.py]: Ah, you meant `is_called_from_pytest`; I meant the function in which this code exists `_knownfailureif`.

**Comment by anntzer** [comment]
Let's get this merged asap?  There's a flaky test it gets rid of :)

**Comment by dstansby** [comment]
Great, going to wait for the rebased tests to pass and then will merge.

**Pull request merged**